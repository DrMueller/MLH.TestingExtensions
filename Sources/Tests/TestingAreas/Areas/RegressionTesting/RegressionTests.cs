using System;
using System.Collections.Generic;
using ApprovalTests;
using ApprovalTests.Reporters;
using Mmu.Mlh.TestingExtensions.Areas.ApprovalTesting;
using Mmu.Mlh.TestingExtensions.FakeApp.Areas.Domain.Models;
using Newtonsoft.Json;
using NUnit.Framework;

namespace Mmu.Mlh.TestingExtensions.Tests.TestingAreas.Areas.RegressionTesting
{
    [TestFixture]
    public class RegressionTests
    {
        [Test]
        [UseReporter(typeof(DiffReporter))]
        public void Regression_MultipleIndviduals()
        {
            var individuals = new List<Individual>
            {
                new Individual("Matthias", "Müller", new DateTime(1986, 12, 29)),
                new Individual("Hans", "Muster", new DateTime(2000, 02, 02))
            };

            ApprovalExtensions.SerializeAndVerifyJson(individuals);
        }

        [Test]
        [UseReporter(typeof(DiffReporter))]
        public void Regression_OneIndvidual()
        {
            var individual = new Individual("Matthias", "Müller", new DateTime(1986, 12, 29));
            ApprovalExtensions.SerializeAndVerifyJson(individual);
        }

        [Test]
        [UseReporter(typeof(DiffReporter))]
        public void Regression_WithScrubber()
        {
            var individual = new Individual("Matthias", "Müller", new DateTime(1986, 12, 29));
            var json = JsonConvert.SerializeObject(individual);
            Approvals.Verify(json, str => str.Replace("Matthias", "Ueli", StringComparison.OrdinalIgnoreCase));
        }
    }
}