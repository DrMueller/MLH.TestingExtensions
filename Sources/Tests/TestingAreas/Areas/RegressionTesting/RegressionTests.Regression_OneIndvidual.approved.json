{
  "BirthDate": "1986-12-29T00:00:00",
  "FirstName": "Matthias",
  "FullName": "Matthias Müller",
  "LastName": "Müller"
}